### PR TITLE
fix the jet response object returned during validate_session_request

### PR DESCRIPTION
**Pull request by guyp-descope**
fix the jet response object returned during validate_session_request so it will now include the adjusted permissions,roles,projectID,userID fields (so it will be just the same as returned from verify_code)

### CODE REVIEW

File: descope/auth.py [hunk 1]
```diff
@@ -355,6 +344,30 @@ def _generate_auth_info(self, response_body: dict, refresh_token: str) -> dict:
                 "tenants", {}
             )
 
+        jwt_response["projectId"] = jwt_response.get(SESSION_TOKEN_NAME, {}).get(
```
Comment (asafshen) [inline descope/auth.py:347]: why do we need to fallback to refresh token's project id and user id?

**Review by asafshen — DISMISSED** [review]
looks good 👍 
added some minor comments

also, making sure this is aligned (or soon to be aligneD) with other sdks logic, right?

File: descope/auth.py [hunk 2]
```diff
@@ -220,7 +220,8 @@ def refresh_token(self, refresh_token: str) -> dict:
         response = self.do_get(uri, None, None, refresh_token)
 
         resp = response.json()
-        return self._generate_auth_info(resp, refresh_token)
+        # return self._generate_auth_info(resp, refresh_token)
```
[the remark this answers is not in the document]
Comment (guyp-descope) [reply descope/auth.py:223]: Yea. Removed.

File: descope/auth.py (hunk 1 above)
Comment (guyp-descope) [reply descope/auth.py:347]: the projectID will be the same in both session token and refresh token so we just take it from the first place we found it.. (its not a new code - I just move it from another place).

**Comment by guyp-descope** [comment]
> looks good 👍 added some minor comments
> 
> also, making sure this is aligned (or soon to be aligneD) with other sdks logic, right?

Its not relevant for other sdk
we choose to have those only in Python.

File: descope/auth.py [hunk 3]
```diff
@@ -321,19 +322,7 @@ def _fetch_public_keys(self) -> None:
...
-            jwt_response[SESSION_TOKEN_NAME] = self._validate_token(st_jwt)
-        rt_jwt = response_body.get("refreshJwt", "")
-        if refresh_token:
-            jwt_response[REFRESH_SESSION_TOKEN_NAME] = self._validate_token(
-                refresh_token
-            )
-        elif rt_jwt:
-            jwt_response[REFRESH_SESSION_TOKEN_NAME] = self._validate_token(rt_jwt)
-
+    def adjust_properties(self, jwt_response: dict):
```
[the remark this answers is not in the document]
Comment (guyp-descope) [reply descope/auth.py:325]: Added inline comments

**Review by asafshen — APPROVED** [review]
👍 👍

**Comment by guyp-descope** [comment]
Because in other languages we have a different structs and the data organized in different way as you may know.

**Pull request merged**